app/src/main/java/com/jerboa/model/LoginViewModel.kt

```diff
@@ -59,9 +57,13 @@ class LoginViewModel : ViewModel() {
                 Toast.makeText(ctx, msg, Toast.LENGTH_LONG).show()
                 return@launch
             }
+            // Login was sucessful after this point
+
+            // Change the lemmy instance to the new one
+            API.changeLemmyInstance(instance, jwt)
```

Comment:
This has some other consequences, there was a reason I put all the way at the end. If the site call fails it will still not allow you to login and force you to retry which is good. For example trying to login with 0.19 now will fail at getSite call but it wont cause you to be logged in. Which would be the case now. The check version check would be even more useless as it would set the client regardless

Reply:
Not sure I follow this one. Right now it is:

- Create a temp API for logging in.
- If the login succeeds, change the API in order to set the JWT.
- Fetch the site to get `MyUserInfo`
- Save it to the DB
- Go home

What am I missing?

Reply:
Fetch Site can fail for multiple reasons.

Currently it fails for incompatible versions.  login succeeds for 0.19 but getSite doesn't in this case you do not want to confirm the api. Currently (with your code) it will display error, login failed but it actually successfully changed the api but it didn't make account. Meaning when ppl back out they are using the new api(for the new account) instead of the actual account that is logged in.

Reply:
Hrm. Shouldn't a failed login be caught in the code just above this?

Reply:
Maybe I should be doing the checks on `tempApi.getSite`, and if those pass, then change the instance and continue with the DB stuff?

Reply:
> Or maybe, login is succeeding, but GetSite fails with older versions?

Yes, not just with older versions also newer versions.

```kt
       API.changeLemmyInstance(instance, jwt)

            // Fetch the site to get your name and id
            siteViewModel.siteRes = apiWrapper(API.getInstance().getSite())

            try {
                when (val siteRes = siteViewModel.siteRes) {
                    is ApiState.Failure -> {
                        val txt = siteRes.msg.message ?: "FAILURE: NO MESSAGE, probably that version not supported"
                        Toast.makeText(ctx, txt, Toast.LENGTH_SHORT).show()
                        throw RuntimeException(txt)
                    }
```

Here we check if the getSite call failed. and show a error msg


The flow when logging with older version:
- Create a temp API for logging in.
- If the login succeeds, change the API in order to set the JWT.
- Confirms the API instance to the new instance
- Fetch the site to get MyUserInfo
- Site Call fails
- Throw error, not supported version, show this
- User backs out the login
- Screen throws errors everywhere, cuz its doing calls with the newly attempted user to login


Before
- Create a temp API for logging in.
- If the login succeeds, change the API in order to set the JWT.
- Fetch the site to get MyUserInfo
- Site Call fails
- Throw error, not supported version, show this
- User backs out the login
- Everything still works because it keeps using the old instance

So just keep the old logic and confirm the api as last as before

Reply:
> Maybe I should be doing the checks on tempApi.getSite, and if those pass, then change the instance and continue with the DB stuff?

We already do this

Reply:
Here's the issue on L63-64 : 

```kt
// Fetch the site to get your name and id
siteViewModel.siteRes = apiWrapper(api.getSite(GetSite(auth)....
```

Its using the tempApi, with a custom JWT, to fetch `MyUserInfo` . That's no longer possible.

I'll see if I can refactor it in a way that works, where it only changes it at the end.

Reply:
Okay I think I got this one now, check the most recent commit or a diff of this file from main.